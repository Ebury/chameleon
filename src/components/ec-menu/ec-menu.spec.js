import { mount } from '@vue/test-utils';
import EcMenu from './ec-menu.vue';

const links = [
  {
    url: '/foo',
    iconName: 'foo-icon',
    text: 'Foo',
  },
  {
    link: '/bar',
    iconName: 'bar-icon',
    text: 'Bar',
  },
  {
    iconName: 'baz-icon',
    text: 'Baz',
  },
  {
    url: '/bat',
    iconName: 'baz-icon',
    text: 'Bat',
  },
];

describe('EcMenu', () => {
  it('should not render if links are not supplied', () => {
    const wrapper = mount(
      EcMenu,
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render if links are null', () => {
    const wrapper = mount(
      EcMenu,
      {
        propsData: {
          links: null,
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render if links is an empty array', () => {
    const wrapper = mount(
      EcMenu,
      {
        propsData: {
          links: [],
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render only links with a url property', () => {
    const wrapper = mount(
      EcMenu,
      {
        propsData: {
          links,
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected when set to horizontal', () => {
    const wrapper = mount(
      EcMenu,
      {
        propsData: {
          horizontal: true,
          links: [links[0]],
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render all items as compact when is horizontal', () => {
    const wrapper = mount(
      EcMenu,
      {
        propsData: {
          horizontal: true,
          links,
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();
    wrapper.findAll('.ec-navigation-link').wrappers.forEach((linkWrapper) => {
      expect(linkWrapper.classes('ec-navigation-link--is-compact')).toBe(true);
    });
  });

  it('should not render all items as compact when horizontal is not set', () => {
    const wrapper = mount(
      EcMenu,
      {
        propsData: {
          horizontal: false,
          links,
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();
    wrapper.findAll('.ec-navigation-link').wrappers.forEach((linkWrapper) => {
      expect(linkWrapper.classes('ec-navigation-link--is-compact')).toBe(false);
    });
  });

  it('should render as expanded by default', () => {
    const wrapper = mount(
      EcMenu,
      {
        propsData: {
          links,
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();

    wrapper.findAll('.ec-navigation-link__text').wrappers.forEach((textWrapper) => {
      expect(textWrapper.isVisible()).toBe(true);
    });
  });

  it('should render as not expanded when isExpanded is passed into', () => {
    const wrapper = mount(
      EcMenu,
      {
        propsData: {
          links,
          isExpanded: false,
        },
      },
    );

    expect(wrapper.element).toMatchSnapshot();

    wrapper.findAll('.ec-navigation-link__text').wrappers.forEach((textWrapper) => {
      expect(textWrapper.isVisible()).toBe(false);
    });
  });
});
