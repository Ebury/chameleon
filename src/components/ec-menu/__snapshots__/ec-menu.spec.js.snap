// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcMenu should not render all items as compact when horizontal is not set 1`] = `
<ul
  class="ec-menu"
>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link"
      href="/foo"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-foo-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
      >
        Foo
      </span>
    </a>
  </li>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link"
      href="/bat"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-baz-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
      >
        Bat
      </span>
    </a>
  </li>
</ul>
`;

exports[`EcMenu should not render if links are not supplied 1`] = `<!---->`;

exports[`EcMenu should not render if links are null 1`] = `<!---->`;

exports[`EcMenu should not render if links is an empty array 1`] = `<!---->`;

exports[`EcMenu should render all items as compact when is horizontal 1`] = `
<ul
  class="ec-menu ec-menu--horizontal"
>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link ec-navigation-link--is-compact"
      href="/foo"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-foo-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
      >
        Foo
      </span>
    </a>
  </li>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link ec-navigation-link--is-compact"
      href="/bat"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-baz-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
      >
        Bat
      </span>
    </a>
  </li>
</ul>
`;

exports[`EcMenu should render as collapsed when isCollapsed is passed into 1`] = `
<ul
  class="ec-menu"
>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link ec-navigation-link--is-collapsed"
      href="/foo"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-foo-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
        style="display: none;"
      >
        Foo
      </span>
    </a>
  </li>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link ec-navigation-link--is-collapsed"
      href="/bat"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-baz-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
        style="display: none;"
      >
        Bat
      </span>
    </a>
  </li>
</ul>
`;

exports[`EcMenu should render as expanded by default 1`] = `
<ul
  class="ec-menu"
>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link"
      href="/foo"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-foo-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
      >
        Foo
      </span>
    </a>
  </li>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link"
      href="/bat"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-baz-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
      >
        Bat
      </span>
    </a>
  </li>
</ul>
`;

exports[`EcMenu should render as expected when set to horizontal 1`] = `
<ul
  class="ec-menu ec-menu--horizontal"
>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link ec-navigation-link--is-compact"
      href="/foo"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-foo-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
      >
        Foo
      </span>
    </a>
  </li>
</ul>
`;

exports[`EcMenu should render only links with a url property 1`] = `
<ul
  class="ec-menu"
>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link"
      href="/foo"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-foo-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
      >
        Foo
      </span>
    </a>
  </li>
  <li
    class="ec-menu__item"
  >
    <a
      class="ec-navigation-link ec-menu__link"
      href="/bat"
    >
      <svg
        class="ec-navigation-link__icon ec-icon"
        height="24"
        width="24"
      >
        <use
          xlink:href="#ec-baz-icon"
        />
      </svg>
       
      <span
        class="ec-navigation-link__text"
        name="ec-navigation-link__text-fade"
      >
        Bat
      </span>
    </a>
  </li>
</ul>
`;
